valgrind.supp

```diff
@@ -634,4 +634,11 @@
    fun:log_record
    fun:*
 }
-
+{
+   From PBSPro (server) - Suppress allocations that clear up when python interpreter restarts
+   Memcheck:Leak
+   fun:malloc
+   fun:PyString_FromString
```

Comment:
@arungrover I don't think this is good to add in suppression file, because as per document of [PyString_FromString](https://docs.python.org/2/c-api/string.html#c.PyString_FromString) it returns new reference which we need to pass to Py_XDECREF (or Py_CLEAR) once we are done with it.

Reply:
@hirenvadalia thanks for reviewing the pull, You are right that PyString_FromString will not be freed unless its reference count comes to 0 but it in this suppression file I've mentioned the function stack to include "pbs_python_populate_attributes_to_python_class" which I think will only be used while preparing input for the hook and will get freed when interpreter restarts.

Have you seen this sort of leak even after interpreter restarts?

Reply:
You are tight @arungrover that suppression you added will be skip leaks if call stack has "pbs_python_populate_attributes_to_python_class".

But "pbs_python_populate_attributes_to_python_class" is PBS' function and PBS prepares input for hook which means it's PBS' responsibility to free that input because PBS is owner of that reference (see first paragraph at [here](https://docs.python.org/2/c-api/intro.html#reference-count-details))

Regarding restart of interpreter:
As you have seen in above link that it is not good idea to restart interpreter (aka call to Py_Initilize and Py_Finalize) multiple time in process as it doesn't guaranty to free all objects which PBS' has allocated, but that is different story...

Answer to you question: Yes you can see that those reference (aka objects) are still live after interpreter restart, but to see that you have to do little hack in PBS' code as below:
Replace  https://github.com/PBSPro/pbspro/blob/master/src/lib/Libpython/pbs_python_external.c#L177
with below:
```
Py_IgnoreEnvironmentFlag = 0;
setenv("PYTHONDUMPREFS", "1", 1);
```
then run your test manually while PBS Server is running in console (with -N) as Python debug logs will not go into PBS' log file.

FYI: There are other variables also if you want to get more debugging info, see [here](https://docs.python.org/2/using/cmdline.html#debug-mode-variables)

Reply:
@arungrover Based on our chat and testing this part, I think for now it is good to add this leak in suppression file till we fix (and remove) restarting of python interpreter so that we get clean valgrind report.